server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java

```diff
@@ -222,12 +222,17 @@ protected void timeoutCheck() throws TransactionException {
         }
         for (GlobalSession globalSession : allSessions) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(
-                    globalSession.getXid() + " " + globalSession.getStatus() + " " + globalSession.getBeginTime() + " "
+                LOGGER.debug(globalSession.getXid() + " " + globalSession.getStatus() + " " + globalSession.getBeginTime() + " "
                         + globalSession.getTimeout());
             }
             boolean shouldTimeout = SessionHolder.lockAndExecute(globalSession, () -> {
-                if (globalSession.getStatus() != GlobalStatus.Begin || !globalSession.isTimeout()) {
+                if (globalSession.getStatus() != GlobalStatus.Begin) {
```

Comment:
If you just find global session of begin status. This would not happen.

Reply:
It can prevent bugs and warn developers.

Reply:
It can prevent bugs and warn developers.